### PR TITLE
tweaks to /etc/localtime management

### PR DESCRIPTION
**Pull request by poettering**
All prompted by https://serverfault.com/questions/991172/invalid-time-zone-utc

### CODE REVIEW

File: src/timedate/timedated.c [hunk 1]
```diff
@@ -277,20 +277,35 @@ static int context_read_data(Context *c) {
...
+         * internal version of UTC in that case behaviour is mostly equivalent. We still prefer creating the
+         * symlink though, since things are more self explanatory then. */
 
-        p = path_join("../usr/share/zoneinfo", c->zone);
-        if (!p)
-                return log_oom();
+        if (isempty(c->zone) || streq(c->zone, "UTC")) {
+
+                if (access("/usr/share/zoneinfo/UTC", F_OK) < 0) {
+
```
Comment (keszybz) [inline src/timedate/timedated.c:292]: All this whitespace makes this seem like more code than it is...

**Comment by keszybz** [comment]
travis timeout in one of the jobs, can be safely ignored.

LGTM.

**Comment by poettering** [comment]
Timeout in some NUMA tests. Let's merge.

**Comment by flokli** [comment]
Can this be backported to `systemd-stable`?

**Comment by keszybz** [comment]
> Can this be backported to `systemd-stable`?

Done now.

**Comment by flokli** [comment]
Thanks!

**Pull request merged**